Vacation-STUDENT_ID_300506334/apps/api/src/admin/vacations/index.ts

```diff
@@ -0,0 +1,176 @@
+import { validateQueryParamsForDeleteVacation } from "./validations";
+import { zeroErrorText } from "../../middlewares/generalValidations";
+import { getUnauthorizedErrorText } from "../../helpers/errorTextFns";
+import { ISignToken } from "../../auth/signTokenFn";
+import { 
+  IUpdateVacation,
+  createVacation, 
+  deleteVacation,
+  getNewVacation,
+  updateVacation, 
+  getIDs,
+  Id,
+  getFollowersPerVacationFullList,
+  IFollowersPerVacationResponseObj,
+} from "./businessLogic";
+
+
+const adminVacationsRouter: Router = Router();
+
+interface TypedRequestQuery<T extends Query> extends Request {
+     currentToken: string;
+     userData: ISignToken;
+    //  userData: any;
+     query: T;
+};
+
+adminVacationsRouter.post(
+  "/",
+  validateCreateVacationBody,
+  createVacationHandler
+  );
+adminVacationsRouter.delete(
+  "/", 
+  deleteVacationHandler
+  );
+adminVacationsRouter.put(
+  "/",
+  validateUpdateVacationBody,
+  updateVacationHandler
+  );
+
+adminVacationsRouter.get(
+"/",
+getFollowersPerVacationFullListHandler
+);
+
+const adminTitle = "admin"
+
+async function updateVacationHandler(
+  req: TypedRequestBody<IUpdateVacation>, 
+  res: Response,
+  next: NextFunction
+  ): Promise<void> {
+    const { role } = req.userData;
+    if (role !== adminTitle) {
+      res.status(401).send(getUnauthorizedErrorText());
+      return;
+    };
+    const { id } = req.body;
+    //vacation id mandatory 
+    console.log(typeof(id), id)
+    if (!id){            // checks if 0 / NaN / ""
+    res.status(422).send(
+    `<<${icons.somethingWentWrong}>> Missing Essential Parameters ${id === 0? `- ${zeroErrorText}` : ""} <<${icons.somethingWentWrong}>>`
+    ); return; }
+    else if (typeof(id) !== typeof(Number(id))) {    // checks if type is different than number
+    res.status(400).send(`<<${icons.warning}>> No Match. Vacation ID Must Be a Number. <<${icons.warning}>>`); return; }
+    const bodyValuesArrIncId = Object.values(req.body)
+    const checkExistingParamsArr = bodyValuesArrIncId.filter(p=> p)
+    console.log(checkExistingParamsArr," checkExistingParamsArr")
+    if ( checkExistingParamsArr.length <= 1) {
+      res.status(400).send(`<<${icons.warning}>> Missing Essential Parameters - Should have Sent At Least One Parameter To Update <<${icons.warning}>>`); return; }
+  try {
+    await updateVacation(req.body, req.userData);
+    res.status(200).send(
+  `<<${icons.registerSuccess}>> Vacation ID No. ${id} Updated successfully <<${icons.registerSuccess}>>`
+  );
+  }
+  catch(ex: unknown){
+    console.log(ex)
+    res.status(403).send(
+      `<<${icons.somethingWentWrong}>> Vacation Failed To Be Updated <<${icons.somethingWentWrong}>>`
+    );
+};
+};
+
+async function createVacationHandler(
+  req: TypedRequestBody<IcreateVacation>, 
+  res: Response,
+  next: NextFunction
+  ): Promise<void> {
+  const { role } = req.userData;
+  if (role !== adminTitle) {
+    res.status(401).send(getUnauthorizedErrorText());
```

Comment:
return  res.status(401).send(getUnauthorizedErrorText());

Reply:
Hi Gal, I have read that -> returning res.status(x).json(x) VS  res.status(x).json(x) ; return; 
is practically doing the same, because the "(req, res)" signature tells express that this is the last function in the chain, and it doesn't expect a return value from that function, so the "return" is basically just "stops" the function. thats why I have decided to go with my option (just to make the code more readable - in my eyes). but I dont mind really to go with your option.

Reply:
Sure - best practice is to stop the function when you return the payload/answer to the client